pkg/api/types.go

```diff
@@ -1042,6 +1042,10 @@ type AzureFileVolumeSource struct {
 	// the ReadOnly setting in VolumeMounts.
 	// +optional
 	ReadOnly bool
+	// the namespace of the secret that contains Azure Storage Account Name and Key
+	// default is the same as the Pod
+	// +optional
+	SecretNameSpace string
```

Comment:
AzureFileVolumeSource is used in pod spec, which cannot specify a cross-namespace secret. Having a field that is forbidden to be set in a pod volumesource context in order to share a struct is sort of odd.